test/test_train.py

```diff
@@ -15,9 +15,7 @@
 
 def train_one_step(model,X,Y):
   params = get_parameters(model)
-  pcount = 0
-  for p in params:
-    pcount += np.prod(p.shape)
+  pcount = sum(np.prod(p.shape) for p in params)
```

Comment:
I'm genuinely curious if there is a performance difference here, or is this purely just more compact?

(ps. I'm just here to learn about how to "neaten" up python code 😄)

Reply:
I did it mostly for the compactness.

But, generally, I think comprehensions are faster than for-loops, which this replaced.

Reply:
Thank you for replying. I might invest some effort in timing some basic comprehensions to replace my for loops to see the difference then :)

Reply:
Go for it.
My general rule:
If you are creating a new object, use comprehensions.
if you are doing a computation, you should stick with for-loops, or use array computations.